### DISCUSSION TITLE
Option to auto-generate subtitles

### DISCUSSION THREAD

**Discussion by KyleMaas** [mediacms-io/mediacms#630] [Feature Requests]
**Describe the feature you'd like**
Having subtitles easily available without manual intervention would be an incredibly helpful feature for the hearing-impaired as well as visual learners.  The way I envision the process for this working is that when you upload a video, you could check a box to optionally take a best guess at subtitles.  When checked, upon upload MediaCMS could use something like this to generate a subtitle file and add it to the media:

https://github.com/abhirooptalasila/AutoSub

It's already under a compatible license and seems to use similar dependencies.

**Describe alternatives you've considered**
Most of the other alternatives I'm finding for auto-generating subtitles use Google as a speech recognition backend.  This is problematic for several reasons including privacy concerns and the need to generate an API key to use it.  By using Mozilla DeepSpeech like AutoSub does above, the subtitles can be generated entirely locally.

**Comment by KyleMaas** [comment]
Bit more information on this program:

https://towardsdatascience.com/generating-subtitles-automatically-using-mozilla-deepspeech-562c633936a7

**Comment by mgogoulos** [comment]
I like the idea and I've been thinking of integrating DeepSpeech for a long time. Whisper has been released very recently and is clearly working very well, I've already prepare a small POC using it and soon plan to open a PR

https://github.com/openai/whisper

**Comment by KyleMaas** [comment]
Very cool.  I've been using AutoSub to good effect for a while now, and it's awfully helpful.  Even if it requires some manual editing it's still a great way to jumpstart the process.

**Comment by homegrownhrbs** [comment]
a whisper integration would be incredible, as well as an editor to quickly fix mistakes.

**Comment by acrawford73** [comment]
Deepgram.com is another option. It is very accurate and can generate WebVTT files. It is a powerful solution to provide each media asset a summary and list of topics, as well as full transcription text of course.
